File: server/routes/submissions.js
const express = require('express')
const { prepare } = require('../database')
const authMiddleware = require('../middleware/auth')

const router = express.Router()

// Submit contact form (public)
router.post('/contact', (req, res) => {
  const { name, email, phone, subject, message } = req.body

  if (!name || !email || !subject || !message) {
    return res.status(400).json({ error: 'Name, email, subject, and message are required' })
  }

  const result = prepare(
    'INSERT INTO submissions (name, email, phone, subject, message) VALUES (?, ?, ?, ?, ?)'
  ).run(name, email, phone || '', subject, message)

  res.status(201).json({
    message: 'Thank you for your message! We will get back to you soon.',
    id: result.lastInsertRowid
  })
})

// Get all submissions (protected)
router.get('/', authMiddleware, (req, res) => {
  const submissions = prepare('SELECT * FROM submissions ORDER BY created_at DESC').all()
  res.json(submissions)
})

// Get single submission (protected)
router.get('/:id', authMiddleware, (req, res) => {
  const submission = prepare('SELECT * FROM submissions WHERE id = ?').get(req.params.id)

  if (!submission) {
    return res.status(404).json({ error: 'Submission not found' })
  }

  res.json(submission)
})

// Mark submission as read (protected)
router.put('/:id/read', authMiddleware, (req, res) => {
  const { id } = req.params

  const existing = prepare('SELECT * FROM submissions WHERE id = ?').get(id)

  if (!existing) {
    return res.status(404).json({ error: 'Submission not found' })
  }

  prepare('UPDATE submissions SET is_read = 1 WHERE id = ?').run(id)

  const updated = prepare('SELECT * FROM submissions WHERE id = ?').get(id)
  res.json(updated)
})

// Delete submission (protected)
router.delete('/:id', authMiddleware, (req, res) => {
  const { id } = req.params

  const existing = prepare('SELECT * FROM submissions WHERE id = ?').get(id)

  if (!existing) {
    return res.status(404).json({ error: 'Submission not found' })
  }

  prepare('DELETE FROM submissions WHERE id = ?').run(id)

  res.json({ message: 'Submission deleted successfully' })
})

// Get stats (protected)
router.get('/stats/overview', authMiddleware, (req, res) => {
  const totalSubmissions = prepare('SELECT COUNT(*) as count FROM submissions').get().count
  const unreadSubmissions = prepare('SELECT COUNT(*) as count FROM submissions WHERE is_read = 0').get().count
  const totalProducts = prepare('SELECT COUNT(*) as count FROM products').get().count

  res.json({
    totalSubmissions,
    unreadSubmissions,
    totalProducts
  })
})

module.exports = router

File: backend/app/api/routes/submissions.py
from fastapi import APIRouter, Depends, HTTPException, status
from sqlalchemy.orm import Session
from sqlalchemy import func
from typing import List

from app.db.session import get_db
from app.models.submission import Submission
from app.models.product import Product
from app.schemas.submission import (
    SubmissionCreate,
    SubmissionResponse,
    SubmissionCreateResponse,
    StatsResponse,
    MessageResponse,
)
from app.core.security import get_current_admin

router = APIRouter(prefix="/submissions", tags=["submissions"])


@router.post("/contact", response_model=SubmissionCreateResponse, status_code=status.HTTP_201_CREATED)
def create_submission(
    submission_data: SubmissionCreate,
    db: Session = Depends(get_db)
):
    """Submit a contact form (public)."""
    if not submission_data.name or not submission_data.email or not submission_data.subject or not submission_data.message:
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST,
            detail={"error": "Name, email, subject, and message are required"}
        )

    submission = Submission(
        name=submission_data.name,
        email=submission_data.email,
        phone=submission_data.phone or "",
        subject=submission_data.subject,
        message=submission_data.message,
    )

    db.add(submission)
    db.flush()
    db.refresh(submission)

    return SubmissionCreateResponse(
        message="Thank you for your message! We will get back to you soon.",
        id=submission.id
    )


@router.get("/stats/overview", response_model=StatsResponse)
def get_stats(
    db: Session = Depends(get_db),
    _: dict = Depends(get_current_admin)
):
    """Get dashboard statistics (protected)."""
    # Total submissions
    total_submissions = db.query(func.count(Submission.id)).scalar() or 0

    # Unread submissions
    unread_submissions = db.query(func.count(Submission.id)).filter(Submission.is_read == 0).scalar() or 0

    # Total products
    total_products = db.query(func.count(Product.id)).scalar() or 0

    return StatsResponse(
        totalSubmissions=total_submissions,
        unreadSubmissions=unread_submissions,
        totalProducts=total_products
    )


@router.get("", response_model=List[SubmissionResponse])
def get_submissions(
    db: Session = Depends(get_db),
    _: dict = Depends(get_current_admin)
):
    """Get all submissions (protected)."""
    submissions = db.query(Submission).order_by(Submission.created_at.desc()).all()
    return submissions


@router.get("/{submission_id}", response_model=SubmissionResponse)
def get_submission(
    submission_id: int,
    db: Session = Depends(get_db),
    _: dict = Depends(get_current_admin)
):
    """Get single submission by ID (protected)."""
    submission = db.query(Submission).filter(Submission.id == submission_id).first()

    if not submission:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail={"error": "Submission not found"}
        )

    return submission


@router.put("/{submission_id}/read", response_model=SubmissionResponse)
def mark_as_read(
    submission_id: int,
    db: Session = Depends(get_db),
    _: dict = Depends(get_current_admin)
):
    """Mark submission as read (protected)."""
    submission = db.query(Submission).filter(Submission.id == submission_id).first()

    if not submission:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail={"error": "Submission not found"}
        )

    submission.is_read = 1
    db.flush()
    db.refresh(submission)

    return submission


@router.delete("/{submission_id}", response_model=MessageResponse)
def delete_submission(
    submission_id: int,
    db: Session = Depends(get_db),
    _: dict = Depends(get_current_admin)
):
    """Delete a submission (protected)."""
    submission = db.query(Submission).filter(Submission.id == submission_id).first()

    if not submission:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail={"error": "Submission not found"}
        )

    db.delete(submission)

    return MessageResponse(message="Submission deleted successfully")
